### ISSUE TITLE
On a LAGG with a load of VLANS

### ISSUE THREAD

**Issue by r3m0dul8** [danb35/freenas-iocage-plex#9]
Anyway I can use this but on a LAGG port that directly connects to VLANs and not to a physical network adapter?

Get this error:

![Untitled](https://user-images.githubusercontent.com/46281217/86539969-e6b4d900-bef8-11ea-8ebf-62cb7c6b1a82.png)

**Comment by r3m0dul8** [comment]
Have also tried modifying the line in plex-jail.sh

`INTERFACE="vnet0"`

to reflect my bridge for the vlan I want it to sit on and it couldn't start the vnet as there was an issue with what I had done to it

**Comment by r3m0dul8** [comment]
![Untitled](https://user-images.githubusercontent.com/46281217/86543593-fbed3000-bf17-11ea-9733-132fc28faa94.png)
Sorry, forgot to add the screenshot

**Comment by r3m0dul8** [comment]
Also, tried installing on the native LAGG physical adapter it it _still_ refuses to work :(
![Untitled](https://user-images.githubusercontent.com/46281217/86544503-5211a180-bf1f-11ea-917a-5b8eddabdda8.png)

**Comment by thorsteneb** [comment]
When you manually create a jail, what options do you use to do that? Checkboxes, interfaces chosen, values, vlan, and so on.

It's quite possible the script needs changes to become vlan-aware.

**Comment by r3m0dul8** [comment]
![Untitled](https://user-images.githubusercontent.com/46281217/87840826-8a0ed200-c899-11ea-8516-98a1583fae04.png)
![Untitled1](https://user-images.githubusercontent.com/46281217/87840828-8aa76880-c899-11ea-94b0-70bd92f2adae.png)
![Untitled2](https://user-images.githubusercontent.com/46281217/87840827-8aa76880-c899-11ea-9cdb-2a5cce719e92.png)




Last one shows the vlan ID it is attached to through a bridge that sits on the LAGG port

This was taken from a active working plex server in TrueNAS 12.1

Hope this helps (sorry for delay)

**Comment by thorsteneb** [comment]
It helps. This will take a bit of testing. The script doesn't support DHCP; does your network allow static IPs?

In the meantime, you can just manually create a jail, run `pkg install plexmediaserver` and `sysrc plexmediaserver_enable="YES"`, mount in your media, and you are up and running. 

If so desired, you could also create an external dataset to hold your configuration data, `rsync -ha` those metadata over from the current jail, run 'mkdir /config' in the new jail, mount the external dataset there, and then as a last step point Plex to that location with `sysrc plexmediaserver_plexpass_support_path="/config"`

**Comment by r3m0dul8** [comment]
Hi

I'm using a PfSense firewall and it doesn't play well with static IP's - a lot of faffing around to get them working and as I control   all network traffic through this, I'd rather just keep it super simple.

Unless there is a way that I can get the MAC address before, then I can add and assign the IP before the jail connects to the network.

**Comment by danb35** [comment]
> I'm using a PfSense firewall and it doesn't play well with static IP's

Nonsense, pfSense plays perfectly well with static IPs (I know, I'm using it).  Just leave some space in your DHCP reservation range (so rather than giving it a range of .2 to .254, give it a range of .100 to .250).  You can set static IPs in .2 to .99.

**Comment by r3m0dul8** [comment]
> > I'm using a PfSense firewall and it doesn't play well with static IP's
> 
> Nonsense, pfSense plays perfectly well with static IPs (I know, I'm using it). Just leave some space in your DHCP reservation range (so rather than giving it a range of .2 to .254, give it a range of .100 to .250). You can set static IPs in .2 to .99.

Granted, so my question would be does setting said static IP range accommodate acme certs and internal DNS assignment?

I've just got too many IP addresses to remember and I prefer to use DNS resolver internally over remembering IPs, they just look so much more fancy too lol

I mean, fair point but is it really so hard to get the script to use DHCP?

**Comment by thorsteneb** [comment]
Enabling DHCP in the script is likely not too difficult. In the meantime, you can very easily do what the script does manually, that’ll be quicker than trying to get the script to work in your specific environment.

**Comment by danb35** [comment]
> does setting said static IP range accommodate acme certs and internal DNS assignment?

Of course it does.  ACME certs aren't affected in the least, and I have no idea where that question would even come from.  Internal DNS assignments are trivial--go to Services -> DNS resolver and add host overrides as needed.

**Comment by r3m0dul8** [comment]
No worries, I'll just stick to the official plug-in. Seems to work fine and without needing to assign anything manually.

I did try to run the commands independently but it didn't want to work - it's why I reported it. Will try what you originally suggested next time (if) I have to do a re-install as I really like the idea to have all the metadata and settings saved elsewhere to prevent having set everything back up again.

Thanks again for the suggestions, even if I didn't quite get round to using it, heard many good things about your repo so wanted to give it a whirl.
